File: src/FestivalPOS/src/api/product.ts
import { Big } from "big.js"
import { bigConverter } from "./converters/big-converter";
import { jsonProperty } from "@utiliread/json"

export class Product {
    @jsonProperty()
    id!: number;

    @jsonProperty()
    name!: string;

    @jsonProperty({ converter: bigConverter })
    price!: Big;

    @jsonProperty()
    previewImageName?: string;

    @jsonProperty()
    previewImageUrl?: string;

    @jsonProperty()
    thumbnailImageName?: string;

    @jsonProperty()
    thumbnailImageUrl?: string;
}

File: src/FestivalPOS/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace FestivalPOS.Models
{
    public class Product
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }
        public decimal Price { get; set; }
        [MaxLength(300)]
        public string PreviewImageName { get; set; }
        public string PreviewImageUrl => PreviewImageName != null ? $"/api/Products/{Id}/Image?kind=preview" : null;
        [MaxLength(300)]
        public string ThumbnailImageName { get; set; }
        public string ThumbnailImageUrl => ThumbnailImageName != null ? $"/api/Products/{Id}/Image?kind=thumbnail" : null;
        public bool IsDeleted { get; set; }
    }
}
